server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -332,28 +497,28 @@ public TestClusterNode currentMaster(ClusterState state) {
             allocationService = ESAllocationTestCase.createAllocationService(settings);
             final ClusterSettings clusterSettings = new ClusterSettings(settings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
             final ThreadPool threadPool = deterministicTaskQueue.getThreadPool();
-            clusterService = new ClusterService(settings, clusterSettings, threadPool, masterService);
-            mockTransport = new DisruptableMockTransport(logger) {
-                @Override
-                protected DiscoveryNode getLocalNode() {
-                    return node;
-                }
-
-                @Override
-                protected ConnectionStatus getConnectionStatus(DiscoveryNode sender, DiscoveryNode destination) {
-                    return ConnectionStatus.CONNECTED;
-                }
-
-                @Override
-                protected Optional<DisruptableMockTransport> getDisruptedCapturingTransport(DiscoveryNode node, String action) {
-                    return Optional.ofNullable(testClusterNodes.nodes.get(node.getName()).mockTransport);
-                }
+            clusterService = new ClusterService(settings, clusterSettings, threadPool, masterService,
+                () -> new PrioritizedEsThreadPoolExecutor(node.getName(), 1, 1, 1, TimeUnit.SECONDS,
```

Comment:
Override the separate threadpool used by the cluster applier service in a very basic way for now.